src/lake/tests/init/test.sh

```diff
@@ -77,6 +77,12 @@ $LAKE -d lean-data exe lean-data
 $LAKE new 123-hello
 $LAKE -d 123-hello exe 123-hello
 
+# Test creating packages both starting with a digit and containing a `.`
+# https://github.com/leanprover/lean4/issues/2999
+
+$LAKE new 123-hello.world
+$LAKE -d 123-hello-world exe 123-hello.world
+
```

Comment:
I am pretty sure this does not test what you think it does. This creates a package with a module named `123Hello.World` located `123Hello/World.lean`. If you want to include the do in the module name, you need to use guillemets (e.g., `lake new «A.B».«C.D»`).

Reply:
That creates a folder name that contains the guillemets!

Reply:
@eric-wieser Oh, yeah. The guillemets in top-level directory name should be stripped (in another PR). Nonetheless, that should create the proper directory structure underneath it (i.e., ` «A-B»-«C-D»/A.B/C.D.lean`) with the fix in this PR.